core/lib/zksync_core/src/genesis.rs

```diff
@@ -419,6 +422,41 @@ pub(crate) async fn save_genesis_l1_batch_metadata(
         .unwrap();
 }
 
+pub(crate) async fn save_set_chain_id_tx(
+    eth_client_url: &str,
+    diamond_proxy_address: Address,
+    state_transition_manager_address: Address,
+    storage: &mut StorageProcessor<'_>,
+) -> anyhow::Result<()> {
+    let eth_client = QueryClient::new(eth_client_url)?;
+    let to = eth_client.block_number("fetch_chain_id_tx").await?.as_u64();
+    let from = to - PRIORITY_EXPIRATION;
+    let filter = FilterBuilder::default()
+        .address(vec![state_transition_manager_address])
+        .topics(
+            Some(vec![SET_CHAIN_ID_EVENT.signature()]),
+            Some(vec![diamond_proxy_address.into()]),
+            None,
+            None,
+        )
+        .from_block(from.into())
+        .to_block(BlockNumber::Latest)
+        .build();
+    let mut logs = eth_client.logs(filter, "fetch_chain_id_tx").await?;
```

Comment:
Dumb question: If `save_set_chain_id_tx` is called prematurely, so that there's no event emitted yet, terminating the node is the best course of action, right?

Reply:
Yes, and since this is only called during genesis, that's not a big deal at all